core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -508,12 +508,8 @@ class KafkaApis(val requestChannel: RequestChannel,
         CompletableFuture.completedFuture(())
       } else if (request.header.apiVersion == 0) {
         // For version 0, always store offsets in ZK.
-        commitOffsetsToZookeeper(
-          request,
-          offsetCommitRequest,
-          authorizedTopicsRequest,
-          responseBuilder
-        )
+        requestHelper.sendMaybeThrottle(request, offsetCommitRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
```

Comment:
I think that we removed version 0 of that API so we could remove the entire branch.